CHANGELOG.md

```diff
@@ -4,7 +4,13 @@
 [Full Changelog](https://github.com/jborgers/sonar-pmd/compare/4.0.3..master)
 
 **Implemented highlights**
-* [TODO]
+* Generate rules xml for the plugin directly from the PMD 7.15.0 rules descriptions: makes all Java and Kotlin rules available and up-to-date automatically
+* Generate nicely formatted html descriptions from the PMD rule description markup
+* Added `pmd` tag and category tag for each rule
+* Added `has-sonar-alternative` tag for rules with known Sonar alternative (instead of making rules with alternatives `Deprecated`)
+* Removed direct access to `XPathRule` and `xpath` property in SonarQube: XPathRule can only be used via rule set definitions
```

Comment:
Because of security? Some people might be using this? Maybe explain the reason?

Reply:
The XPathRule "exposed as rule" was already Deprecated, not sure why. I can imagine it is confusing when this rule represents many other rules with their own severity, estimated fix time, documentation. Better to define an explicit PMD rule implemented as xpath, which can have all of these attributes by itself.

Reply:
I now see it was a "Template Rule", that way you **can** generate new rules with their own attributes.